controllers/cassandrabackup_controller_test.go

```diff
@@ -0,0 +1,279 @@
+package controllers
+
+import (
+	"context"
+	"fmt"
+	"reflect"
+	"strconv"
+	"time"
+
+	"github.com/k8ssandra/medusa-operator/pkg/medusa"
+	"github.com/k8ssandra/medusa-operator/pkg/pb"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	cassdcapi "github.com/datastax/cass-operator/operator/pkg/apis/cassandra/v1beta1"
+	api "github.com/k8ssandra/medusa-operator/api/v1alpha1"
+)
+
+const (
+	TestCassandraDatacenterName = "dc1"
+	timeout                     = time.Second * 10
+	interval                    = time.Millisecond * 250
+)
+
+var _ = Describe("CassandraBackup controller", func() {
+	i := 0
+	testNamespace := ""
+
+	BeforeEach(func() {
+		testNamespace = "backup-test-" + strconv.Itoa(i)
+		testNamespace := &corev1.Namespace{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: testNamespace,
+			},
+		}
+		Expect(k8sClient.Create(context.Background(), testNamespace)).Should(Succeed())
+		i = i + 1
+	})
+
+	Specify("create a new backup for an existing CassandraDatacenter", func() {
+		By("create a CassandraDatacenter")
+		backupName := "test-backup"
+
+		cassdcKey := types.NamespacedName{
+			Name:      TestCassandraDatacenterName,
+			Namespace: testNamespace,
+		}
+
+		cassdc := &cassdcapi.CassandraDatacenter{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:        cassdcKey.Name,
+				Namespace:   cassdcKey.Namespace,
+				Annotations: map[string]string{},
+			},
+			Spec: cassdcapi.CassandraDatacenterSpec{
+				ClusterName:   "test-dc",
+				ServerType:    "cassandra",
+				ServerVersion: "3.11.7",
+				Size:          3,
+				StorageConfig: cassdcapi.StorageConfig{
+					CassandraDataVolumeClaimSpec: &corev1.PersistentVolumeClaimSpec{
+						VolumeName: "data",
+					},
+				},
+			},
+		}
+		Expect(k8sClient.Create(context.Background(), cassdc)).Should(Succeed())
+		Eventually(func() bool {
+			created := &cassdcapi.CassandraDatacenter{}
+			err := k8sClient.Get(context.Background(), cassdcKey, created)
+			return err == nil
+		}, timeout, interval).Should(BeTrue())
+
+		By("create the datacenter service")
+		dcServiceKey := types.NamespacedName{Namespace: cassdcKey.Namespace, Name: cassdc.GetAllPodsServiceName()}
+		dcService := &corev1.Service{
+			ObjectMeta: metav1.ObjectMeta{
+				Namespace: dcServiceKey.Namespace,
+				Name:      dcServiceKey.Name,
+			},
+			Spec: corev1.ServiceSpec{
+				Selector: map[string]string{
+					cassdcapi.ClusterLabel: cassdc.Spec.ClusterName,
+				},
+				Ports: []corev1.ServicePort{
+					{
+						Name: "cql",
+						Port: 9042,
+					},
+				},
+			},
+		}
+		Expect(k8sClient.Create(context.Background(), dcService)).Should(Succeed())
+		Eventually(func() bool {
```

Comment:
These could be ``func() error ....Should(BeNil())``. No need to cast to boolean.